packages/components/src/slot-fill/fill.js

```diff
@@ -21,14 +21,23 @@ function FillComponent( { name, children, registerFill, unregisterFill } ) {
 
 	useLayoutEffect( () => {
 		registerFill( name, ref.current );
+		// Ignore reason: The ref.current value will likely change before the cleanup is run.
+		// This may be intentional, but could be impacted by future refactors of the other useLayoutEffects in this component.
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I may be getting confused by similar conversation in other PRs, but is there a particular reason why we're not solving this warning by storing the ref in a local variable? It looks like a relatively safe change:

<details>

<summary>Click to see suggested changes</summary>

```diff
diff --git a/packages/components/src/slot-fill/fill.js b/packages/components/src/slot-fill/fill.js
index eda7fe16d4..583b62ba27 100644
--- a/packages/components/src/slot-fill/fill.js
+++ b/packages/components/src/slot-fill/fill.js
@@ -20,11 +20,10 @@ function FillComponent( { name, children, registerFill, unregisterFill } ) {
 	} );
 
 	useLayoutEffect( () => {
-		registerFill( name, ref.current );
-		// Ignore reason: The ref.current value will likely change before the cleanup is run.
-		// This may be intentional, but could be impacted by future refactors of the other useLayoutEffects in this component.
-		// eslint-disable-next-line react-hooks/exhaustive-deps
-		return () => unregisterFill( name, ref.current );
+		const refValue = ref.current;
+		registerFill( name, refValue );
+
+		return () => unregisterFill( name, refValue );
 		// Ignore reason: the useLayoutEffects here are written to fire at specific times, and introducing new dependencies could cause unexpected changes in behavior.
 		// We'll leave them as-is until a more detailed investigation/refactor can be performed.
 		// eslint-disable-next-line react-hooks/exhaustive-deps
```

</details>

Reply:
Initially, I wasn't sure if it would really be a safe change, because the two subsequent layout effects update the ref value before the cleanup runs (`ref.current..children` and `ref.current.name` respectively).

I do see your point though, looking at it again. The useLayoutEffect in question specifically fires only on the first render... so when the next two layout effects fire they won't actually be changing anything. On future re-renders when the `name` or `children` values may have been updated, this initial layout effect won't fire at all due to the empty dep array, so changes won't be a problem. Updated in c4aed99

Reply:
To me, the point is that we want to make sure that `unregisterFill` is called with the same `refValue` that was used to `registerFill`.

Anyway, changes LGTM. Feel free to merge !